cedar-policy-core/src/ast/name.rs

```diff
@@ -324,20 +342,22 @@ impl std::fmt::Display for SlotId {
     }
 }
 
-/// Two possible variants for Slots
-#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+/// Three possible variants for Slots
+#[derive(Debug, Clone, Eq, PartialEq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 pub(crate) enum ValidSlotId {
     #[serde(rename = "?principal")]
     Principal,
     #[serde(rename = "?resource")]
     Resource,
+    GeneralizedSlot(Id),
```

Comment:
Would be good to have a doc comment here explaining what a GeneralizedSlot is, maybe link to the RFC